src/common/components/DropZone.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+
+import { useDropzone } from 'react-dropzone';
+import { useTranslation } from 'react-i18next';
+
+import { Alert, Paper, Stack, Typography } from '@mui/material';
+
+import theme from 'theme';
+
+const getFormatedSize = bytes => {
+  const size = bytes.toLocaleString(undefined, {
+    maximumFractionDigits: 2,
+  });
+  return `${size}`;
+};
+
+const CurrentFile = ({ file }) => {
+  const size = getFormatedSize(file.size / 1000.0);
+  return (
+    <Typography sx={{ fontWeight: 'bold' }}>
+      {file.name} {size}KB
+    </Typography>
+  );
+};
+
+const DropZone = props => {
+  const { t } = useTranslation();
+  const {
+    maxSize,
+    maxFiles,
+    fileExtensions,
+    multiple,
+    onDrop,
+    onDropRejected,
+    currentFile,
+    errors,
+    className,
+  } = props;
+  const accept = fileExtensions.map(extension => `.${extension}`).join(',');
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({
+    onDrop,
+    onDropRejected,
+    accept,
+    useFsAccessApi: false,
+    multiple,
+    maxSize,
+    maxFiles,
+  });
+
+  return (
+    <Stack
+      className={className}
+      component={Paper}
+      square
+      direction="column"
+      alignItems="center"
+      justifyContent="center"
+      spacing={1}
+      variant="outlined"
+      sx={({ palette }) => ({
+        backgroundColor: isDragActive ? palette.blue.mid : palette.blue.lite,
+        border: `2px dashed ${palette.blue.dark}`,
+        paddingTop: errors ? 0 : 2,
+        paddingBottom: 3,
+        minHeight: '125px',
+        cursor: 'pointer',
+      })}
+      {...getRootProps()}
+    >
+      <input {...getInputProps()} />
+      {isDragActive ? (
+        <Typography>{t('common.drop_zone_drop_message')}</Typography>
+      ) : (
+        <>
+          {errors &&
+            errors.map((error, index) => (
+              <Alert
+                key={index}
+                style={{
+                  width: '100%',
+                  boxSizing: 'border-box',
+                  margin: `0 0 ${theme.spacing(1)}`,
+                }}
+                severity="error"
+              >
+                {error}
+              </Alert>
+            ))}
+          <Paper
+            variant="outlined"
+            sx={({ spacing, palette }) => ({
+              padding: `${spacing(1)} ${spacing(3)}`,
+              borderColor: palette.black,
+            })}
+          >
+            {t('common.drop_zone_select_file')}
+          </Paper>
+          <Typography
+            variant="caption"
+            sx={{ fontWeight: 'bold', paddingTop: 1 }}
+          >
+            {t('common.drop_zone_format', {
+              extensions: fileExtensions.map(ext => `*.${ext}`).join(', '),
```

Comment:
question: should we call `map()` outside the component and place the value in a `const`?

Reply:
@paulschreiber Can you add more context on why you think that will be better? Me reasoning here was:
This is only specifically needed for `common.drop_zone_format`, so there is no value on having the joined extensions string far from the localized string parameters, I think having it close can potentialy let us improve this (I am not sure if I like the join here completely, but that can be a future improvement), and the scope of action of this is really clear.